apps/aechannel/src/aesc_fsm.erl

```diff
@@ -4160,6 +4186,42 @@ handle_call(St, Req, From, #data{} = D) ->
 handle_call(_St, _Req, From, D) ->
     keep_state(D, [{reply, From, {error, unknown_request}}]).
 
+handle_call_(awaiting_signature, cancel_update, From, #data{} = D) ->
+    #data{op = #op_sign{tag = Tag}} = D,
+    %% cancelable tags for an action initiated by me:
+    CancelMyTags = [ slash_tx
+                   , deposit_tx
+                   , withdraw_tx
+                   , snapshot_solo_tx
+                   , close_solo_tx
+                   , settle_tx
+                   , ?UPDATE
+                   , ?SHUTDOWN],
+    %% cancelable tags for an action initiated by the other party:
+    CancelAckTags = [ ?DEP_CREATED
+                    , ?WDRAW_CREATED
+                    , ?UPDATE_ACK
+                    , ?SHUTDOWN_ACK],
+    case {lists:member(Tag, CancelMyTags), lists:member(Tag, CancelAckTags)} of
+        {true, _} ->
+            report(info, canceled_update, D),
+            lager:debug("update canceled", []),
+            next_state(open, clear_ongoing(D#data{op = ?NO_OP}),
+                      [{reply, From, ok}]);
+        {_, true} ->
+            report(info, canceled_update, D),
+            lager:debug("update canceled", []),
+            handle_recoverable_error(#{ code => ?ERR_ABORT
+                                      , respond => true
+                                      , msg_type => Tag }, D, [{reply, From, ok}],
+                                     _ReportConflictToClient = false);
+        {false, false} ->
+            lager:debug("update can not be canceled for ~p", [Tag]),
+            keep_state(D, [{reply, From, {error, not_allowed_now}}])
+    end;
+handle_call_(_NonSigningState, cancel_update, From, #data{} = D) ->
+    lager:debug("update can not be canceled while ~p", [_NonSigningState]),
+    keep_state(D, [{reply, From, {error, not_allowed_now}}]);
```

Comment:
The error tag is somewhat weird. How about `invalid_state`?

Reply:
Well the state is actually valid, it is that the action is not allowed at the current state. The actual message propagated to the client is `Not allowed at current channel state` as described [here](https://github.com/aeternity/aeternity/blob/0d73c39ed467bb68b62a1b42af57f97d732addba/apps/aehttp/src/sc_ws_api_jsonrpc.erl#L262)